compiler/noirc_evaluator/src/ssa/parser/tests.rs

```diff
@@ -633,3 +633,22 @@ fn test_parses_keyword_in_function_name() {
         ";
     assert_ssa_roundtrip(src);
 }
+
+#[test]
+fn regression_modulo_fields_brillig() {
+    use crate::brillig::BrilligOptions;
+    use std::panic::AssertUnwindSafe;
+
+    let src = "
+        brillig(inline) predicate_pure fn main f0 {
+          b0(v0: Field, v1: Field):
+            v2 = mod v0, v1
+            return v2
+        }
+        ";
+    let ssa = Ssa::from_str(src).unwrap();
+    assert!(
+        std::panic::catch_unwind(AssertUnwindSafe(|| ssa.to_brillig(&BrilligOptions::default())))
+            .is_err()
+    );
+}
```

Comment:
```suggestion
#[test]
#[should_panic = "Attempt to modulo fields"]
fn regression_modulo_fields_brillig() {
    use crate::brillig::BrilligOptions;
    use std::panic::AssertUnwindSafe;

    let src = "
        brillig(inline) predicate_pure fn main f0 {
          b0(v0: Field, v1: Field):
            v2 = mod v0, v1
            return v2
        }
        ";
    let ssa = Ssa::from_str(src).unwrap();
    ssa.to_brillig(&BrilligOptions::default());
}
```

I would write this test like this. If compiling from ssa to brillig succeeds (i.e. we're doing a bad modulo) then the test will fail. If the test panics with a different message then the test will fail. The test will only succeed if we hit the panic with the message "Attempt to modulo fields".